src/toady/resolve_service.py

```
@@ -82,6 +93,15 @@ def resolve_thread(self, thread_id: str) -> Dict[str, Any]:
 
         except ValueError as e:
             raise ResolveServiceError(f"Invalid thread ID: {e}") from e
+        except GitHubAPIError as e:
+            # Handle specific GraphQL errors
+            if "not found" in str(e).lower():
+                raise ThreadNotFoundError(f"Thread {thread_id} not found") from e
+            elif "permission" in str(e).lower() or "forbidden" in str(e).lower():
+                raise ThreadPermissionError(
+                    f"Permission denied to resolve thread {thread_id}"
+                ) from e
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "raise", remove the leading "el" from "elif"

(R1720)